freqtrade/strategy/interface.py

```diff
@@ -275,7 +298,7 @@ def get_signal(self, pair: str, interval: str,
             logger.warning('Empty dataframe for pair %s', pair)
             return False, False
 
-        latest = dataframe.iloc[-1]
+        latest = dataframe.loc[dataframe['date'] == latest_date].iloc[-1]
```

Comment:
Odd question - do we need the `.iloc[-1]` here? 

the only case where `dataframe.loc[dataframe['date'] == latest_date]` would return more than one line is if i drop one line - and duplicate the last line so i got the same date twice ... (or am i mistaking...)?

Reply:
if you don't add iloc, it will return a dataframe, so it would need some other changes below but haven't checked, anyway the functions only check specific rows, so it could be possible that the returned df might have the same dates in between also considering that the dates are not the index

Reply:
ah right, it's a datatype issue - didn't think about that :+1: